### DISCUSSION TITLE
Customize file info

### DISCUSSION THREAD

**Discussion by Lassebq** [gokcehan/lf#1282] [Ideas]
There should be a way to modify what the bottom line displays. Even ranger has some color in it. (file permissions display in cyan) while in lf it's just gray string.
I would like to make it the same as exa's output:
![image](https://github.com/gokcehan/lf/assets/68742864/6ef68fae-4845-4149-9aca-5e229639cda3)

**Comment by joelim-work** [comment]
Currently there's no support for configuring this.

I guess it would be possible to add a `fileinfofmt` option similar to the `promptfmt` option, where you can specify what fields you want displayed and insert terminal sequences for style/coloring. Though I'm not sure whether it's worth supporting different colors for each letter in the permission flags.

**Reply by Lassebq** [reply]
Allow it to be set to the output of a program/shell script? while passing in selected file path. Similarly to previewer

**Reply by joelim-work** [reply]
There was a similar discussion in #1257 when I was working on customising the status bar (the bottom right corner of the screen). Running an external program/shell script each time a file is selected isn't instantaneous and can cause a small and sometimes noticeable delay when drawing the UI. This is the case for the file previews, at least on my setup.

In the ideal world, everything would be able to customised by delegated to an external program. But performance has to be considered too.

**Reply by ilyagr** [reply]
I'm also leaning against making this configurable. At least, I wouldn't want to do it myself. The way @joelim-work suggested seems best, but I'm not a fan of how complex `promptfmt` is and it doesn't seem fun to implement another mini-language like it does.

However, I'll mention that if you're OK with compiling a Go program, it'd be quite easy to colorize the status bar by editing the format string mentioned in https://github.com/gokcehan/lf/issues/1103#issuecomment-1459725705 and recompiling. You would need to use the same ANSI escape sequences as in `promptfmt`. Making different permissions letters have different colors would be a bit harder, but no harder than making an external script doing it.

**Reply by joelim-work** [reply]
To be clear, I don't mind making this configurable by adding a `fileinfofmt` option as this has been suggested in #1103 before. The only thing I'm against is implementing something complex like applying styles to _part of a field_ (e.g. a separate color for each letter in the file permissions).

I'm thinking of copying the [default](https://github.com/ranger/ranger/blob/1a007158cef1279df68d0e8567b126def6b68046/ranger/colorschemes/default.py#L109-L111) from `ranger` and use `set fileinfofmt "\033[36m%p\033[0m %c %u %g %s %t %l"` where:

- `%p` is the file permissions
- `%c` is the link count
- `%u` is the user
- `%g` is the group
- `%s` is the size
- `%t` is the last modified time
- `%l` is the link target including the `->` arrow (if exists)

**Reply by ilyagr** [reply]
If you're up for doing it, sounds good!

I have already thought of a couple of very minor quibbles:

For the link target, I like the `->` that lf prints. I suppose we could have a version of `%l` that includes it.

Subjectively, I'm not sure about changing the default to color the permissions. I don't think it will necessarily look better. To me, it seems relevant the user enters commands on the same line, and the command-line doesn't have color. If we color the permissions, OTOH, maybe we should also color the command-line prompt symbol (`>`,`$`, etc.).

(The point about color is not a hill I'm willing to die on, though, and I might even change my mind.)

--------------

As an aside, the only things I'd want to change for myself (I'm not suggesting doing this by default) is to get rid of the link count and maybe the group.

**Reply by joelim-work** [reply]
Oh yeah I meant for `%l` to include the arrow, I have amended the description above. I don't think it's worth customising the behaviour of the arrow for now, format strings in general have trouble handling text that is conditional on the presence of other fields.

Regarding the defaults, I don't think it matters too much anyway since the user will be able to customise it.

**Reply by ilyagr** [reply]
Re arrow: if we really want to, we could have `%l` that doesn't have the arrow and `%L` that does include it. Both of these would be empty if the file is not a link.
